api/v1alpha1/issuer_interface.go

```diff
@@ -28,4 +28,5 @@ type Issuer interface {
 	runtime.Object
 	metav1.Object
 	GetStatus() *IssuerStatus
+	GetIssuerTypeIdentifier() string
```

Comment:
This new function should return the prefix used in SignerName, which often has this format "<resources>.<group>". This is an example: "simpleclusterissuers.issuer.cert-manager.io".

It might be necessary to revisit this API in the future & make it harder to make mistakes here.

Reply:
I would prefer if you add this information as a comment to the func.

Reply:
As we discussed, the reason we are not exposing the whole CertificateRequest must be explained as a comment above the interface. The reason we need to document it is that it is a surprising choice.

Reply:
Woah, great job on that comment!!! It reads really well. 🙌